org.eclipse.xtext/src/org/eclipse/xtext/formatting2/regionaccess/internal/AbstractEObjectRegion.java

```diff
@@ -84,7 +87,7 @@ public EStructuralFeature getContainingFeature() {
 		if (feature == null) {
 			return null;
 		}
-		return semanticElement.eClass().getEStructuralFeature(feature);
+		return semanticElement.eContainer().eClass().getEStructuralFeature(feature);
```

Comment:
It looks like this code is working solely on the AST level. Why can't we use EObject.eContainingFeature() ?

Reply:
In case where AST and the regionaccess gets out of sync, we prefer to return `null` instead of the probably wrong containment feature.

Reply:
In that case, a guard could make sense, to check if the containingFeature and the feature name from the grammar are the same. What do you think?

Smth like semanticElement.eContainingFeature.name == feature?